test/e2e_test.go

```diff
@@ -251,31 +257,31 @@ func TestGCSStorage(t *testing.T) {
 
 	client, err := storage.NewClient(ctx)
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
 	bucketName, rmBucket := makeBucket(t, client)
-	defer rmBucket()
+	t.Cleanup(rmBucket)
 
 	c, ns, cleanup := setup(ctx, t, setupOpts{})
-	defer cleanup()
+	t.Cleanup(cleanup)
 
 	resetConfig := setConfigMap(ctx, t, c, map[string]string{
 		"artifacts.taskrun.signer":  "x509",
 		"artifacts.taskrun.storage": "gcs",
 		"storage.gcs.bucket":        bucketName,
 	})
-	defer resetConfig()
+	t.Cleanup(resetConfig)
 	time.Sleep(3 * time.Second)
```

Comment:
possibly another sleep that can be removed?

Reply:
commented